### DISCUSSION TITLE
Modifying the Audio Kit Keys - Success

### DISCUSSION THREAD

**Discussion by Slider2732** [marcel-licence/esp32_midi_sampler#15] [Show and tell]
Marcel's latest video shows how to remove and replace some resistors on the Audio Kit V2.2 board.
Doing so enables the onboard 6 Keys to have full functionality.
Video here: https://www.youtube.com/watch?v=r0af0DB1R68

I only had through hole 'regular' resistors, which were 1.5K rather than 1.8K
Tried the conversion anyway  :)
It works!
Pic of the conversion: https://gofile.io/d/WNVclW
(IMG_3679.JPG)

Here's the test code:

---------------------------------------
int reading;

void setup() 
{
  pinMode(36, INPUT);
  Serial.begin(115200);
}

void loop() 
{
 reading = analogRead(36); 
 Serial.println(reading);
}

---------------------------------------

With the following readings, based on multiple presses:
No key pressed - 4095
Key 1 - 0
Key 2 - 368-384
Key 3 - 794-800
Key 4 - 1125-1132
Key 5-- 1389-1398
Key 6 - 1618-1625

A decent enough range between each Key.
Hope it helps those who may have some trepidation with the conversion. It can look like mine and still work!

**Comment by marcel-licence** [comment]
Wow good work! Thanks for the picture. I was really curious to see it. Other values will also work. I tried just to get the most stable result. My code is also online now. I tried to get it a bit more stable to avoid glitches by reading voltage transitions. You have really good readings.

**Comment by Slider2732** [comment]
It went well, once I realised to bend the legs to the right shape and cut them. 
The resistors were from the same batch, part of a multi resistor values box. Each tested within a few ohms of each other. 

Just seen your upload... and downloaded the new version. Cannot wait to try it  :)

**Reply by marcel-licence** [reply]
It looks like that your solution was less painful than mine xD
The SMD components didn't want to get contact with the soldering pads below because there was no space for the solder to flow underneath.

**Comment by Slider2732** [comment]
I tend to use a frowned on method for SMD. The pad is dabbed with solder flux, then an older solder tip is used which has scrapes etc on it (or can use rough sandpaper on the tip of a new one). Solder will then stick to it and form a small bead on the end. I put the bead to the SMD part that's sitting on the pad and the solder will flow fine. The flux makes for a good finish. 
A more correct method, is to reflow the pads with solder, so it sits higher off the board. As in, put a bit extra on. When the SMD part is put on top and held with tweezers, the solder underneath melts.

The code works great. I turned off the constant VT100 output for the Arduino serial monitor and enabled the Keys serial output line. A bit of figuring out by watching the screen while pressing saw how to step through the functions. 
My keyboard is in another room to the PC, so it was still hit and miss with no screen. Have loaded samples and did get it to record via the mic. The recording went something like 'click, background noise, click, "don't think it worked, hmm", click, background noise, click, click' LOL. Which, I then played up and down the keyboard.   

A possible issue, is that files seem to be somewhat random when they load. Luckily perhaps for your knowledge of a problem, is my own files run from 1.wav to 99.wav and are split into banks of 10 by sound type. The new code will load something like 6.wav, 4.wav, 5.wav, 11.wav, 3.wav, 14.wav. The MMC driver might be looking for letters?  
It's no problem if people have half a dozen samples and they're named for what they are, such as lowstrings.wav, but could be a problem when stepping through a larger organised bank. My own changes incremented a variable by 1 to load the next sample, the driver code will do it a different way. 

Good luck with the screens code... can't wait for that either  :)
An I2C SSD1306 was tried again, with Adafruit library but it still stopped when the AC101 setup started. I'll have to be patient!

**Comment by jazzbiddy** [comment]
@Slider2732 Is the link to your picture working still? I couldn't see one when I visited the site.

@marcel-licence Is the mod to change the keys required? What functions will not be available if I don't do the mod?  If I do the mod what do I get. Thanks!

**Reply by marcel-licence** [reply]
The mod is required if you need more IOs and/or want to use the buttons at the same time. The board comes with the buttons connected to multiple IOs which are shared also with the pin header. In addition to that the mod is required to allow using SPI/I2C via the pin header of the audio kit. For normal operation (MIDI only) no modification is required.

**Reply by jazzbiddy** [reply]
My plan right now is to just use midi to play sounds, so it seems like I should be able to keep things as they are. Thanks for the info.

**Comment by Slider2732** [comment]
Sorry for the late reply. 
It does look like the file was deleted.... thanks for pointing it out. 
All it actually showed, was 'regular' through hole resistors with bent little legs. I used those as I had no SMD dinky tiny ones. The resistors look like an 'n' shape, with the legs cut short. 

Will try to upload it here - 


![IMG_3679](https://user-images.githubusercontent.com/8705560/126581207-67914450-14fa-42bd-83b2-aa1c6558670d.JPG)

**Reply by jazzbiddy** [reply]
Got it. thanks for uploading that again. It looks like I don't need to do the mod, right now, but with this along with Marcel's video - I think I can do it if I need to!
